src/main/java/frc/robot/subsystems/Drivebase.java

```diff
@@ -4,15 +4,148 @@
+  private CANSparkMax leftLeader = new CANSparkMax(DrivebaseConstants.LEFT_LEADER, MotorType.kBrushless);
+  private CANSparkMax leftFollower = new CANSparkMax(DrivebaseConstants.LEFT_FOLLOWER, MotorType.kBrushless);
+
+  private CANSparkMax rightLeader = new CANSparkMax(DrivebaseConstants.RIGHT_LEADER, MotorType.kBrushless);
+  private CANSparkMax rightFollower = new CANSparkMax(DrivebaseConstants.RIGHT_FOLLOWER, MotorType.kBrushless);
+
+  private RelativeEncoder m_rightEncoder;
+  private RelativeEncoder m_leftEncoder;
+
+  // The left-side drive encoder
+  Encoder leftEncoders = new Encoder(DrivebaseConstants.LEFT_LEADER, DrivebaseConstants.LEFT_FOLLOWER, false, Encoder.EncodingType.k2X);
+
+  // The right-side drive encoder
+  Encoder rightEncoders = new Encoder(DrivebaseConstants.RIGHT_LEADER, DrivebaseConstants.RIGHT_FOLLOWER, false, Encoder.EncodingType.k2X);
+
+  private double leftEncoderStart, rightEncoderStart;
+
+  private AHRS gyro;
+
+  private double leftStart;
+  
+  private double rightStart;
+
+  private DifferentialDrive diffDrive = new DifferentialDrive(leftLeader, rightLeader);
+
+  private final edu.wpi.first.math.kinematics.DifferentialDriveOdometry odometry;
+
+  public Drivebase() {
+    leftFollower.follow(leftLeader);
+    rightFollower.follow(rightLeader);
+
+    leftLeader.setIdleMode(IdleMode.kBrake);
+    leftFollower.setIdleMode(IdleMode.kBrake);
+    rightLeader.setIdleMode(IdleMode.kBrake);
+    rightFollower.setIdleMode(IdleMode.kBrake);
+
+    leftLeader.setInverted(true);
+    leftFollower.setInverted(true);
+
+    rightLeader.setInverted(true);
+    rightFollower.setInverted(true);
+
+    m_leftEncoder = leftLeader.getEncoder();
+    m_rightEncoder = rightLeader.getEncoder();
+
+    leftStart = m_leftEncoder.getPosition();
+    rightStart = m_rightEncoder.getPosition();
+
+    resetEncoders();
+
+    leftLeader.setSmartCurrentLimit(DrivebaseConstants.LEFT_DRIVE_GROUP_CURRENT_LIMIT);
+    rightLeader.setSmartCurrentLimit(DrivebaseConstants.RIGHT_DRIVE_GROUP_CURRENT_LIMIT);
+
+    leftFollower.setSmartCurrentLimit(DrivebaseConstants.LEFT_DRIVE_GROUP_CURRENT_LIMIT);
+    rightFollower.setSmartCurrentLimit(DrivebaseConstants.RIGHT_DRIVE_GROUP_CURRENT_LIMIT);
+
+    // Configures the encoder to return a distance of 4 for every 256 pulses
+    // Also changes the units of getRate  
+    leftEncoders.setDistancePerPulse(4./256.);
+    rightEncoders.setDistancePerPulse(4./256.);
+
+    gyro = new AHRS(SPI.Port.kMXP);
+
+    leftEncoderStart = leftEncoders.getDistance();
+    rightEncoderStart = rightEncoders.getDistance();
+
+    odometry = new edu.wpi.first.math.kinematics.DifferentialDriveOdometry(gyro.getRotation2d());
+
+    diffDrive.setDeadband(0.02);
+  }
+
+  public void curvatureDrive(double speed, double rotation, boolean quickTurn) {
+    diffDrive.curvatureDrive(speed, rotation, quickTurn);
+  }
+
+  //ENCODER STUFF
+  public double getLeftWheelEncoder() {
```

Comment:
Is there a reason we're tracking the last encoder reset position instead of just setting the position to zero?

Reply:
it was in 2021-Offseason and I thought there was a specific reason for it. Git blame said that @JakeBoggs added it, @JakeBoggs was there a reason?